src/flag_gems/ops/div.py

```diff
@@ -77,22 +77,52 @@ def trunc_divide(A, B):
         return A / B
 
 
+@triton.jit
+def _int_floordiv(x, y):
+    # TODO: request Triton to add an integer remainder builtin
+    # The semantic of Triton floordiv differs from Pytorch/Numpy
+    # Triton floordiv equates to
+    #     (x - np.fmod(x, y)) / y
+    # whereas Pytorch floordiv is
+    #     (x - np.remainder(x, y)) y
+    # The results show a one off difference when
+    #     C1) x and y have opposite signs
+    # and C2) x is not multiples of y.
+    # Apart from the above, there's an erroneous case x // 0 returns -1
+    # whereas in Pytorch x // 0 returns -1 if x >=0 and -2 if x < 0
+    # but this special case is coalesced into the c1 and c2 check so
+    # there's extra handling.
```

Comment:
Mod by 0 and division by zero is somehow undefined, so many all languages are free to define it without consistency.

Also, triton has a bug that when `x//y` and `x%y` appears together, `x%y` may have different result when `y==0`.

```python
import triton
import torch
from triton import language as tl

from flag_gems.utils import pointwise_dynamic

@pointwise_dynamic(promotion_methods=[(0, 1, "DEFAULT"), (0, 1, "DEFAULT")])
@triton.jit
def f(x, y):
    return x // y, x % y

@pointwise_dynamic(promotion_methods=[(0, 1, "DEFAULT")])
@triton.jit
def g(x, y):
    return x // y

@pointwise_dynamic(promotion_methods=[(0, 1, "DEFAULT")])
@triton.jit
def h(x, y):
    return x % y

x = torch.tensor([2, -2, 0], dtype=torch.int32, device="cuda")
y = torch.tensor([2, -2, 0], dtype=torch.int32, device="cuda").unsqueeze(-1)

out = f(x, y)

print("x // y,  x % y")
print(out[0])
print(out[1])

print("x // y")
print(g(x, y))

print("x % y")
print(h(x, y))
```

outputs
```text
x // y,  x % y
tensor([[ 1, -1,  0],
        [-1,  1,  0],
        [-1, -1, -1]], device='cuda:0', dtype=torch.int32)
tensor([[ 0,  0,  0],
        [ 0,  0,  0],
        [ 2, -2,  0]], device='cuda:0', dtype=torch.int32)
x // y
tensor([[ 1, -1,  0],
        [-1,  1,  0],
        [-1, -1, -1]], device='cuda:0', dtype=torch.int32)
x % y
tensor([[ 0,  0,  0],
        [ 0,  0,  0],
        [-1, -1, -1]], device='cuda:0', dtype=torch.int32)
```

Reply:
I will issue a bug report to triton-lang with an example without using pointwise dynamic.

Reply:
This makes the situation more complicated...